python/lsst/pex/logging/debug.cc

```diff
@@ -24,22 +24,35 @@
 
 #include "lsst/pex/logging/Debug.h"
 
-using namespace lsst::pex::logging;
-
 namespace py = pybind11;
+using namespace pybind11::literals;
+
+namespace lsst {
+namespace pex {
+namespace logging {
 
-PYBIND11_PLUGIN(_debug) {
-    py::module mod("_debug", "Access to the classes from the pex logging Debug library");
+PYBIND11_PLUGIN(debug) {
+    py::module mod("debug");
 
-    py::class_<Debug> cls(mod, "Debug", py::base<Log>());
+    py::class_<Debug, std::shared_ptr<Debug>, Log> cls(mod, "Debug");
 
-    cls.def(py::init<const std::string&, int>(), py::arg("name"),
-            py::arg("verbosity") = -1 * Log::INHERIT_THRESHOLD);
+    cls.attr("default_max_debug") = py::none();
 
-    cls.def(py::init<const Log&, const std::string&, int>(), py::arg("parent"), py::arg("name"),
-            py::arg("verbosity") = -1 * Log::INHERIT_THRESHOLD);
+    cls.def("__init__", [&cls](Debug & instance, const std::string & name, py::object verbosity) {
+            verbosity = verbosity.is_none() ? cls.attr("default_max_debug") : verbosity;
+            new (&instance) Debug(name, verbosity.is_none() ? -1 * Log::INHERIT_THRESHOLD : py::cast<int>(verbosity));
```

Comment:
I'm curious why all this was needed here and in the next constructor? If it is done to compute the value of the `verbosity` argument then I suspect it would be simpler to make a local function such as the following:
```
int getVerbosity(py::object verbosity, Class & cls) {
    verbosity = verbosity.is_none() ? cls.attr("default_max_debug") : verbosity;
    return  verbosity.is_none() ? -1 * Log::INHERIT_THRESHOLD : py::cast<int>(verbosity);
}
```
and use that as the default value: `"verbosity"_a = getVerbosity(verbosity, cls)` (if one *can* use a function like this).

Note that I have assumed `using Class = py::class_<Debug, std::shared_ptr<Debug>, Log>;` which I think would be worth adding if you do adopt this solution.